src/v/cluster/archival/async_data_uploader.cc

```diff
@@ -190,11 +190,12 @@ segment_upload::make_segment_upload(
   inclusive_offset_range range,
   size_t read_buffer_size,
   ss::scheduling_group sg,
-  model::timeout_clock::time_point deadline) {
+  model::timeout_clock::time_point deadline,
+  bool allow_unstable_reads) {
```

Comment:
>This is useful in situations where the desired upload bounds are a priori known
>to be safe but may not precisely match the consumer visible offset. e.g. when
>doing time-based uploads, direct segment collection, or in test.

I thought for time-based uploads and direct segment collection, we'd still want to respect the Raft majority-replicated offset (because offsets above this point can be suffix truncated and rewritten). Can you elaborate on a sequence of events where we'd want to upload above the Raft visible offset?

Reply:
At the time, I saw some weird behavior in some test and observed that we don't go through raft to construct readers in the old code at all, instead going straight to disk w/ some file offsets. 

Circumventing the visible offset clamping in `raft::make_reader` seemed to unstick whatever the test was, and since, as Evgeny points out, the archiver is LSO driven anyway, clamping the reader config seems unnecessary for correctness.

With all that said, I'm going to drop this commit here and reintroduce it if needed w/ a better investigation attached. All of this code is inert at this point anyway, so there's no immediate need.